### DISCUSSION TITLE
How to training yolov8 include slice aided hyper inference (sahi) for deploying model to android

### DISCUSSION THREAD

**Discussion by rahmadyd** [ultralytics/ultralytics#15435] [Q&A]
excuse me, i want to ask about How to training **yolov8 include slice aided hyper inference (sahi)** because i want to export the weight but i dont know where i put sahi code if i export that weight to tensorflowlite (i want to deploy this model to android). please tell me how it works. thanks

**Comment by glenn-jocher** [comment]
@rahmadyd to train YOLOv8 with SAHI for sliced inference and then export to TensorFlow Lite for Android deployment, you first train your model normally with YOLOv8. SAHI is used during inference, not training, so you don't need to modify your training process. After training, you can export your model to TensorFlow Lite using the `export` function. For integrating SAHI during inference on Android, you would handle the slicing and stitching logic in your Android application code. For detailed steps, please refer to our documentation: https://docs.ultralytics.com/guides/sahi-tiled-inference/.

**Reply by agoransson** [reply]
I've been looking for that answer for very long - I've previously understood that SAHI requires special training logic, to slice training images in the same way as images are sliced during inference. You're saying this isn't needed, "any" Yolo object-detection model should do fine while running inference with SAHI?

Thank you.

**Reply by glenn-jocher** [reply]
@agoransson that's correct—SAHI is used only during inference to handle large images by slicing them into smaller tiles. Any YOLO model trained on standard datasets should work fine with SAHI for inference without requiring special training logic.

**Reply by rygall** [reply]
If you're training models a model on unique high-resolution images to detect objects that are not in standard datasets, is there a standard built-in method to integrate SAHI into training?

**Reply by glenn-jocher** [reply]
@rygall great question! SAHI is designed for inference and doesn't require special training integration. For high-resolution training data, we recommend standard YOLO training with your full images. If you want to experiment with sliced training data augmentation, you'd need to implement that as a custom preprocessing step. The SAHI inference integration remains unchanged - see our SAHI Tiled Inference guide for implementation details.

**Reply by rygall** [reply]
My only problem with the standard YOLO training is that the images appear to be resized down to 640x640 (I believe) and then a lot of the small details that are needed for the contextual understanding of why a bounding box exists, aren't visible. Am I not understanding that properly?

**Reply by glenn-jocher** [reply]
For high-resolution training, you can adjust the `imgsz` parameter (e.g., `imgsz=1280`) and enable mosaic augmentation in your YOLO training configuration to better preserve small object details. While images are resized during training, these adjustments help maintain contextual information. For implementation details, see the Ultralytics YOLO Training documentation and SAHI Tiled Inference guide for sliced inference strategies.

**Reply by agoransson** [reply]
How does one apply mosaic augmentation on the Ultralytics HUB? I can see a custom command input, but I'm unsure how to use that. Thank you for always being very fast @glenn-jocher at tips and tricks.

**Reply by glenn-jocher** [reply]
To apply mosaic augmentation in Ultralytics HUB, use the `close_mosaic` parameter in your training arguments or custom commands field. For example, add `close_mosaic=0` to disable mosaic disabling in final epochs. The mosaic augmentation is enabled by default with our recommended settings - no special configuration needed unless modifying default behavior. For implementation details, see the Mosaic augmentation documentation.

**Comment by agoransson** [comment]
A follow-up, I've tried to find a way to run sliced inference on device. My initial tests have all failed - so I am wondering. Is there any similar technique to detect small objects available for tflite models?

**Reply by glenn-jocher** [reply]
@agoransson while there isn't a direct equivalent to SAHI (Slicing Aided Hyper Inference) specifically for TFLite models, you can explore exporting your model to TFLite with INT8 quantization for optimized performance on edge devices, as described in the [TFLite - Ultralytics YOLO Docs](https://docs.ultralytics.com/integrations/tflite/). You might also consider optimizing at the application level on your Android device for small object detection.
